app/src/main/java/com/vero/tasky/agenda/presentation/agenda/AgendaViewModel.kt

```diff
@@ -83,16 +94,33 @@ class AgendaViewModel @Inject constructor(
         updateState(getInitialState(date = day))
     }
 
-    private fun getInitialState(date: LocalDate) : AgendaState {
+    private fun getInitialState(date: LocalDate): AgendaState {
         val days = LocalDateParser.getNextDaysFromDate(date)
         return AgendaState(
             userShortName = UserNameParser.toShortName(preferences.getUser()?.fullName ?: ""),
             monthLabel = date.month.getDisplayName(TextStyle.FULL, Locale.getDefault()),
             days = days,
-            dayLabel = LocalDateParser.getDayLabel(days[0]),
+            dayLabel = LocalDateParser.getDayLabel(days[3].date),
         )
     }
 
+    private fun setCurrentAgendaItem(date: LocalDate, agendaItems: List<AgendaItem>) {
+        if (date != LocalDate.now())
+            updateState(state.copy(currentAgendaItem = null))
+        else {
+            val currentDayTime = LocalDateTime.now()
+            var currentAgendaItem: AgendaItem? = null
+            agendaItems.forEach { agendaItem ->
+                if (currentDayTime >= agendaItem.time)
+                    currentAgendaItem = agendaItem
+                else
+                    return@forEach
+            }
```

Comment:
I'm a friend of using all the helpful functions from the Kotlin standard lib. In this case you could simply do this:
```
val currentAgendaItem = agendaItems.find { it.time <= currentDayTime }
```